geoutils/vector/vector.py

```diff
@@ -64,37 +63,105 @@ class Vector:
+            ds = gpd.read_file(filename_or_dataset)
         # If GeoPandas or Shapely object is passed
         elif isinstance(filename_or_dataset, (gpd.GeoDataFrame, gpd.GeoSeries, BaseGeometry)):
-            self._name = None
             if isinstance(filename_or_dataset, gpd.GeoDataFrame):
-                self._ds = filename_or_dataset
+                ds = filename_or_dataset
             elif isinstance(filename_or_dataset, gpd.GeoSeries):
-                self._ds = gpd.GeoDataFrame(geometry=filename_or_dataset)
+                ds = gpd.GeoDataFrame(geometry=filename_or_dataset)
             else:
-                self._ds = gpd.GeoDataFrame({"geometry": [filename_or_dataset]}, crs=None)
-        # If Vector is passed, simply point back to Vector
-        elif isinstance(filename_or_dataset, Vector):
-            for key in filename_or_dataset.__dict__:
-                setattr(self, key, filename_or_dataset.__dict__[key])
-            return
+                ds = gpd.GeoDataFrame({"geometry": [filename_or_dataset]}, crs=None)
         else:
-            raise TypeError("Filename argument should be a string, Path or geopandas.GeoDataFrame.")
+            raise TypeError("Filename argument should be a string, path or geodataframe.")
+
+        # Set geodataframe
+        self.ds = ds
+
+        # Write name attribute
+        if isinstance(filename_or_dataset, str):
+            self._name = filename_or_dataset
+        if isinstance(filename_or_dataset, pathlib.Path):
+            self._name = filename_or_dataset.name
+
+    @property
+    def crs(self) -> CRS:
+        """Coordinate reference system of the vector."""
+        return self.ds.crs
+
+    @property
+    def ds(self) -> gpd.GeoDataFrame:
+        """Geodataframe of the vector."""
+        return self._ds
+
+    @ds.setter
+    def ds(self, new_ds: gpd.GeoDataFrame | gpd.GeoSeries) -> None:
+        """Set a new geodataframe."""
+
+        if isinstance(new_ds, gpd.GeoDataFrame):
+            self._ds = new_ds
+        elif isinstance(new_ds, gpd.GeoSeries):
+            self._ds = gpd.GeoDataFrame(geometry=new_ds)
+        else:
+            raise ValueError("The dataset of a vector must be set with a GeoSeries or a GeoDataFrame.")
+
+    def vector_equal(self, other: gu.Vector, **kwargs: Any) -> bool:
+        """
+        Check if two vectors are equal.
+
+        Keyword arguments are passed to geopandas.assert_geodataframe_equal.
+        """
+
+        try:
+            assert_geodataframe_equal(self.ds, other.ds, **kwargs)
+            vector_eq = True
+        except AssertionError:
+            vector_eq = False
+
+        return vector_eq
+
+    @property
+    def name(self) -> str | None:
+        """Name on disk, if it exists."""
+        return self._name
+
+    @property
+    def geometry(self) -> gpd.GeoSeries:
+        return self.ds.geometry
+
+    @property
+    def columns(self) -> pd.Index:
```

Comment:
Should it not be consistent with the `PointCloud` class, which uses `all_columns`?